### PR TITLE
Replace Travis CI configuration with GitHub Actions

### PR DESCRIPTION
**Pull request by akx**
This PR removes the defunct Travis CI configuration and replaces it with a near-minimal GitHub Actions workflow.

There are some things still left undone (e.g. testing on Postgres again, and testing against the current django `main` (née `master`) revision, but I think this is a good start; I added relevant `TODO` markers to follow up on.

It also adds testing on Django 4.1 and 4.2.

(Following this, I think the next version of `django-polymorphic` should drop testing and support on EOL versions of Python and Django.)

### CODE REVIEW

**Comment by j-antunes** [comment]
@akx - Thanks for you work on this PR. Just to confirm. This would replace the work in GHA that was started on #558?

**Comment by akx** [comment]
@j-antunes Yes, #560 + this effectively replace #558, but this uses the current Tox configuration to test against various Django and Python versions, which didn't happen in #558.

**Comment by j-antunes** [comment]
Sounds good! Everything looks good on the PR. I'll wait for @AdamDonna to check it, to have a second set of eyes on it.

Again, thanks for you work!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - master
```
Comment (hugovk) [inline .github/workflows/ci.yml:10]: This lets us test feature branches on forks, and also trigger builds via the GitHub UI:

```suggestion
on: [push, pull_request, workflow_dispatch]
```

File: .github/workflows/ci.yml
Comment (hugovk) [inline .github/workflows/ci.yml:1]: Let's call the file `test.yml`, makes it more like other Jazzband files, and easier to maintain.

https://github.com/search?q=org%3Ajazzband+path%3A**%2Ftest.yml+language%3AYAML&type=code

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+jobs:
+  Build:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
```
Comment (hugovk) [inline .github/workflows/ci.yml:16]:
```suggestion
      - uses: actions/checkout@v4
      - uses: actions/setup-python@v5
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          - os: ubuntu-22.04
+            python-version: "3.8"
+          - os: ubuntu-22.04
+            python-version: "3.9"
+          - os: ubuntu-22.04
+            python-version: "3.10"
+    steps:
+      - uses: actions/checkout@v3
+      - name: "Set up Python ${{ matrix.python-version }}"
+        uses: actions/setup-python@v4
```
Comment (hugovk) [inline .github/workflows/ci.yml:48]:
```suggestion
      - uses: actions/checkout@v4
      - name: "Set up Python ${{ matrix.python-version }}"
        uses: actions/setup-python@v5
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    branches:
+      - master
+jobs:
+  Build:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.12"
```
Comment (hugovk) [inline .github/workflows/ci.yml:18]:
```suggestion
          python-version: "3"
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
+name: CI
```
Comment (hugovk) [inline .github/workflows/ci.yml:1]:
```suggestion
name: Test
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  push:
+    branches:
+      - master
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - master
+jobs:
+  Build:
```
Comment (hugovk) [inline .github/workflows/ci.yml:12]: Move this job into a `release.yml`

https://github.com/search?q=org%3Ajazzband+path%3A**%2Frelease.yml+language%3AYAML&type=code

File: README.rst [hunk 7]
```diff
@@ -1,5 +1,5 @@
-.. image::  https://travis-ci.org/jazzband/django-polymorphic.svg?branch=master
-    :target: http://travis-ci.org/jazzband/django-polymorphic
+.. image::  https://github.com/jazzband/django-polymorphic/actions/workflows/ci.yml/badge.svg
+    :target: https://github.com/jazzband/django-polymorphic/actions/workflows/ci.yml
```
Comment (hugovk) [inline README.rst:2]:
```suggestion
.. image::  https://github.com/jazzband/django-polymorphic/actions/workflows/test.yml/badge.svg
    :target: https://github.com/jazzband/django-polymorphic/actions/workflows/test.yml
```

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+          - os: ubuntu-20.04
+            python-version: "3.6"
+          - os: ubuntu-20.04
+            python-version: "3.7"
+          - os: ubuntu-22.04
+            python-version: "3.8"
+          - os: ubuntu-22.04
+            python-version: "3.9"
+          - os: ubuntu-22.04
+            python-version: "3.10"
```
Comment (hugovk) [inline .github/workflows/ci.yml:44]: Would be good to drop EOL Python versions and use a regular matrix, but that can be a followup 👍

Comment (akx) [reply .github/workflows/ci.yml]: Yeah, that was mentioned in #556.

**Review by hugovk — APPROVED** [review]

**Comment by akx** [comment]
I'll just go ahead and merge this since there's 2 approvals 👍

**Pull request merged**